File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.java
class Solution {
    public int maxIceCream(int[] costs, int coins) {
         Arrays.sort(costs);

        final int[] numCoins = new int[]{coins};

        int numIceCreamBars = Math.toIntExact(Arrays.stream(costs)
                .filter(cost -> cost <= numCoins[0])
                .peek(cost -> numCoins[0] -= cost) 
                .count()); 

        return numIceCreamBars;
    }
}

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()
        noOfBars = 0
        for cost in costs:
            if (cost > coins): 
                break
            coins -= cost
            noOfBars += 1
        return noOfBars